src/autoload/Pyegi.lua

```diff
@@ -168,16 +157,140 @@ function Main(sub, sel)
+					actor_exists = false
+					for _, s_t in pairs(actors_table) do
+						if s_t == sub[l_t].actor then actor_exists = true end
+					end
+					if not actor_exists and sub[l_t].actor ~= "" then table.insert(actors_table,sub[l_t].actor) end
+				end
+				if actors_table then
+					inputs_gui={
+						{x=0,y=0,class="label",label="Style to apply on:"},
+						{x=1,y=0,class="dropdown",name="apply_on_style",items=styles_table,value=styles_table[1]},
+						{x=2,y=0,class="label",label="Actor to apply on:"},
+						{x=3,y=0,class="dropdown",name="apply_on_actor",items=actors_table,value=actors_table[1]}
+					}
+				else
+					inputs_gui={
+						{x=0,y=0,class="label",label="Style to apply on:"},
+						{x=1,y=0,class="dropdown",name="apply_on_style",items=styles_table,value=styles_table[1]},
+						{x=2,y=0,class="label",label="No actors to display!"}
+					}
+				end
+			elseif res.apply_on2 == "Style or Actor" then
+				for _, l_t in ipairs(desired_lines) do
+					style_exists = false
+					for _, s_t in pairs(styles_table) do
+						if s_t == sub[l_t].style then style_exists = true end
+					end
+					if not style_exists then table.insert(styles_table,sub[l_t].style) end
+					actor_exists = false
+					for _, s_t in pairs(actors_table) do
+						if s_t == sub[l_t].actor then actor_exists = true end
+					end
+					if not actor_exists and sub[l_t].actor ~= "" then table.insert(actors_table,sub[l_t].actor) end
+				end
+				if actors_table then
+					inputs_gui={
+						{x=0,y=0,class="label",label="Style to apply on:"},
+						{x=1,y=0,class="dropdown",name="apply_on_style",items=styles_table,value=styles_table[1]},
+						{x=2,y=0,class="label",label="Actor to apply on:"},
+						{x=3,y=0,class="dropdown",name="apply_on_actor",items=actors_table,value=actors_table[1]}
+					}
+				else
+					inputs_gui={
+						{x=0,y=0,class="label",label="Style to apply on:"},
+						{x=1,y=0,class="dropdown",name="apply_on_style",items=styles_table,value=styles_table[1]},
+						{x=2,y=0,class="label",label="No actors to display!"}
+					}
+				end
+			end
+		end
 		for _, items1 in pairs(py_settings.Controls) do
-			table.insert(inputs_gui,{class=items1.class,name=items1.name,x=items1.x,y=items1.y,width=items1.width,height=items1.height,label=items1.label,hint=items1.hint,text=items1.text,value=items1.value,min=items1.min,max=items1.max,step=items1.step,items=items1.items})
+			table.insert(inputs_gui,{class=items1.class,name=items1.name,x=items1.x,y=items1.y+1,width=items1.width,height=items1.height,label=items1.label,hint=items1.hint,text=items1.text,value=items1.value,min=items1.min,max=items1.max,step=items1.step,items=items1.items})
 		end
+		APT("")
 		local btn2, res2 = ADD(inputs_gui, btns2, {ok=btns2[1], cancel=btns2[-1]})
 		
-		if btn2 == btns[1] then
+		if btn2 == btns[1] then -- "Apply"
+			-- modifying the desired lines' number of the table
+			local desired_lines_modified = {}
```

Comment:
why "modified" and not just "desired_lines"?
also for the comment, I think "-- getting the desired lines" is better

Reply:
The desired_lines variable is defined already and is being used in this loop of generating desired_lines_modified variable. I wanted to name it desired_lines2 but it didn't seem cool (^_^;)

Reply:
hmm. I still can't map its name to its application. what do you think about `filtered_desired_lines`?